pkg/syncer/cnsoperator/util/util.go

```diff
@@ -140,3 +149,28 @@ func GetTKGVMIP(ctx context.Context, vmOperatorClient client.Client, dc dynamic.
 	log.Infof("Found external IP Address %s for VirtualMachine %s/%s", ip, vmNamespace, vmName)
 	return ip, nil
 }
+
+// GetNetworkProvider reads the network-config configmap in Supervisor cluster and
+// returns the network provider as NSX-T or VDS. Returns an error if network provider
+// is not present in the configmap.
+func GetNetworkProvider(ctx context.Context) (string, error) {
+	log := logger.GetLogger(ctx)
+	k8sclient, err := k8s.NewClient(ctx)
+	if err != nil {
+		log.Errorf("GetNetworkProvider: Creating Kubernetes client failed. Err: %v", err)
+		return "", err
+	}
+	cm, err := k8sclient.CoreV1().ConfigMaps(kubeSystemNamespace).Get(ctx, wcpNetworkConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get config map %q in namespace %q", wcpNetworkConfigMap, kubeSystemNamespace)
```

Comment:
Can you also print the error that's being returned?

Reply:
To clarify, i mean `err` on line 164 can be added to the log/return

Reply:
Why not just return the error to the caller?

Reply:
@RaunakShah done.